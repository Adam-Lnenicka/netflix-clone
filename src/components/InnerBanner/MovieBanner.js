import React from "react";
import { useParams } from "react-router";
import { useApiMoviesArraySelector } from "../../store/selectors";
import { ButtonExit } from "../../styles/global/Buttons";
import Shorten from "../Shorten";
import {
  MovieContainer,
  MovieDetailsImage,
  MovieRating,
  Runtime,
} from "./styles/MovieBannerStyles";

const MovieBanner = () => {
  const { movieId } = useParams();
  const apiMoviesArray = useApiMoviesArraySelector();
  const thisMovie = apiMoviesArray.map((movie) =>
    movie.find((movie) => movie.id == movieId)
  );

  return (
    <MovieContainer>
      <MovieDetailsImage
        src={thisMovie.map((m) => m.poster_path)}
        alt={thisMovie.map((m) => m.title)}
        onError={(e) =>
          (e.target.src =
            "https://images.unsplash.com/photo-1560109947-543149eceb16?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1035&q=80")
        }
      />
      <div>
        <div className="flex">
          <ButtonExit to="/">X</ButtonExit>
          <h2> {thisMovie.map((m) => m.title)}</h2>
          <MovieRating> {thisMovie.map((m) => m.vote_average)}</MovieRating>
        </div>
        <p>{thisMovie.map((m) => m.tagline)}</p>
        <Runtime>
          <p> {Shorten(thisMovie.map((m) => m.release_date))} </p>
          <p> {thisMovie.map((m) => m.runtime)} min</p>
        </Runtime>
        <p>{thisMovie.map((m) => m.overview)}</p>
      </div>
    </MovieContainer>
  );
};

export default MovieBanner;
