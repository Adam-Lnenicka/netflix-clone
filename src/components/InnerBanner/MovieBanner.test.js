import "jsdom-global/register";

import React from "react";
import { mount, shallow } from "enzyme";
import MovieBanner from "./MovieBanner";
import { Provider } from "react-redux";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";

describe("MovieBanner", () => {
  const initialState = {};
  const mockStore = configureStore();
  let store;
  let props;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);

    props = {
      thisMovie: () => {},
      apiMoviesArray: ["test, test2, test3"],
      title: "Captain Marvel",
      genres: ["Animation"],
      release_date: "2016",
      runtime: 108,
      vote_average: 7.7,
      overview:
        "Carol DanversDetermined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery. becomes one of the universe`s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races.",
      poster_path: "https://",
      tagline: "short description",
    };
    wrapper = shallow(
      <Provider store={store}>
        <MovieBanner {...props} />
      </Provider>
    );
  });

  it("MovieBanner to match snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Objects empty by default", () => {
    const store = mockStore(initialState);

    expect(wrapper.find("#banner-selected-image")).toEqual({});
  });
});
