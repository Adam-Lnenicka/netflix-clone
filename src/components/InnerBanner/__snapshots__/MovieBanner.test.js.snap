// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieBanner MovieBanner to match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <MovieBanner
    apiMoviesArray={
      Array [
        "test, test2, test3",
      ]
    }
    genres={
      Array [
        "Animation",
      ]
    }
    overview="Carol DanversDetermined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery. becomes one of the universe\`s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races."
    poster_path="https://"
    release_date="2016"
    runtime={108}
    tagline="short description"
    thisMovie={[Function]}
    title="Captain Marvel"
    vote_average={7.7}
  />
</ContextProvider>
`;
